### ISSUE TITLE
Dask dashboard produces 500 error when using 128 threads or more

### ISSUE THREAD

**Issue by albarji** [dask/distributed#8307]
**Describe the issue**:

When creating a Dask cluster using 128 workers or more, accessing the Dask dashboard through a browser returns a "500: Internal Server Error".

**Minimal Complete Verifiable Example**:

If you have a large enough machine you can try this locally with

```python
from distributed import LocalCluster

cluster = LocalCluster(n_workers=128, threads_per_worker=1)
```
and then connecting through a browser to `http://localhost:8787`.

This can also be tested by creating 2 x2idn.16xlarge instances on AWS, with one worker per vCPU, for a total of 128 workers. Looking at the scheduler logs I get the following messages:

```
2023-10-26 06:37:51,729 - bokeh.application.handlers.function - ERROR - operands could not be broadcast together with shapes (40,) (2,)
--
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/distributed/utils.py", line 748, in wrapper
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/components/scheduler.py", line 4237, in status_doc
workers_memory.update()
File "/usr/local/lib/python3.9/site-packages/bokeh/core/property/validation.py", line 95, in func
return input_function(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/components/scheduler.py", line 631, in update
update(self.source, d)
File "/usr/local/lib/python3.9/site-packages/bokeh/core/property/validation.py", line 95, in func
return input_function(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/utils.py", line 50, in update
if source.data == data:
ValueError: operands could not be broadcast together with shapes (40,) (2,)
2023-10-26 06:37:51,730 - tornado.application - ERROR - Uncaught exception GET /status (10.0.2.42)
HTTPServerRequest(protocol='http', host='localhost:8787', method='GET', uri='/status', version='HTTP/1.1', remote_ip='10.0.2.42')
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/tornado/web.py", line 1704, in _execute
result = await result
File "/usr/local/lib/python3.9/site-packages/bokeh/server/views/doc_handler.py", line 54, in get
session = await self.get_session()
File "/usr/local/lib/python3.9/site-packages/bokeh/server/views/session_handler.py", line 144, in get_session
session = await self.application_context.create_session_if_needed(session_id, self.request, token)
File "/usr/local/lib/python3.9/site-packages/bokeh/server/contexts.py", line 243, in create_session_if_needed
self._application.initialize_document(doc)
File "/usr/local/lib/python3.9/site-packages/bokeh/application/application.py", line 194, in initialize_document
h.modify_document(doc)
File "/usr/local/lib/python3.9/site-packages/bokeh/application/handlers/function.py", line 143, in modify_document
self._func(doc)
File "/usr/local/lib/python3.9/site-packages/distributed/utils.py", line 748, in wrapper
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/components/scheduler.py", line 4237, in status_doc
workers_memory.update()
File "/usr/local/lib/python3.9/site-packages/bokeh/core/property/validation.py", line 95, in func
return input_function(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/components/scheduler.py", line 631, in update
update(self.source, d)
File "/usr/local/lib/python3.9/site-packages/bokeh/core/property/validation.py", line 95, in func
return input_function(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/utils.py", line 50, in update
if source.data == data:
ValueError: operands could not be broadcast together with shapes (40,) (2,)
2023-10-26 06:37:52,133 - bokeh.application.handlers.function - ERROR - operands could not be broadcast together with shapes (40,) (2,)
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/distributed/utils.py", line 748, in wrapper
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/components/scheduler.py", line 4237, in status_doc
workers_memory.update()
File "/usr/local/lib/python3.9/site-packages/bokeh/core/property/validation.py", line 95, in func
return input_function(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/components/scheduler.py", line 631, in update
update(self.source, d)
File "/usr/local/lib/python3.9/site-packages/bokeh/core/property/validation.py", line 95, in func
return input_function(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/utils.py", line 50, in update
if source.data == data:
ValueError: operands could not be broadcast together with shapes (40,) (2,)
2023-10-26 06:37:52,133 - tornado.application - ERROR - Uncaught exception GET /status (10.0.2.42)
HTTPServerRequest(protocol='http', host='localhost:8787', method='GET', uri='/status', version='HTTP/1.1', remote_ip='10.0.2.42')
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/tornado/web.py", line 1704, in _execute
result = await result
File "/usr/local/lib/python3.9/site-packages/bokeh/server/views/doc_handler.py", line 54, in get
session = await self.get_session()
File "/usr/local/lib/python3.9/site-packages/bokeh/server/views/session_handler.py", line 144, in get_session
session = await self.application_context.create_session_if_needed(session_id, self.request, token)
File "/usr/local/lib/python3.9/site-packages/bokeh/server/contexts.py", line 243, in create_session_if_needed
self._application.initialize_document(doc)
File "/usr/local/lib/python3.9/site-packages/bokeh/application/application.py", line 194, in initialize_document
h.modify_document(doc)
File "/usr/local/lib/python3.9/site-packages/bokeh/application/handlers/function.py", line 143, in modify_document
self._func(doc)
File "/usr/local/lib/python3.9/site-packages/distributed/utils.py", line 748, in wrapper
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/components/scheduler.py", line 4237, in status_doc
workers_memory.update()
File "/usr/local/lib/python3.9/site-packages/bokeh/core/property/validation.py", line 95, in func
return input_function(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/components/scheduler.py", line 631, in update
update(self.source, d)
File "/usr/local/lib/python3.9/site-packages/bokeh/core/property/validation.py", line 95, in func
return input_function(*args, **kwargs)
File "/usr/local/lib/python3.9/site-packages/distributed/dashboard/utils.py", line 50, in update
if source.data == data:
ValueError: operands could not be broadcast together with shapes (40,) (2,)
```

**Anything else we need to know?**:

I suspect the problem is due to holoviews, so here are the versions of the related packages

```
holoviews == 1.17.0
bokeh == 2.4.3
param == 1.13.0
```

**Environment**:

- Dask version: 2022.11.1
- Python version: 3.9.18
- Operating System: Ubuntu 22.04.1
- Install method (conda, pip, source): pip

**Comment by jrbourbeau** [comment]
Thanks for the issue @albarji. I recall seeing this traceback before and think it's been fixed recently. I see you're on an older version of Dask. Could you upgrade `dask` and `distributed` to the latest release and see if you encounter the same issue?

**Comment by albarji** [comment]
@jrbourbeau thanks for the fast reply. I have updated to `dask==2023.10.1` and `distributed==2023.10.1` and I can confirm the problem disappears in the minimal example I provided above.

Unfortunately I'm working in a large system using Dask, and updating to the latest release introduces some incompatibilities. Do you happen to know at which dask/distributed release was this bug solved? That would be very helpful to reduce the number of issues that tend to pop up when updating to a very different version.

**Comment by jrbourbeau** [comment]
> I have updated to dask==2023.10.1 and distributed==2023.10.1 and I can confirm the problem disappears in the minimal example I provided above.

Okay, glad to hear it

> Do you happen to know at which dask/distributed release was this bug solved? That would be very helpful to reduce the number of issues that tend to pop up when updating to a very different version.

Yeah, that makes sense. I _think_ it was the `2023.6.1` release with https://github.com/dask/distributed/pull/7932, but unfortunately, I'm not 100% certain about that. That fix was actually for `numpy=1.25` support. What version of `numpy` is installed? Another alternative may be to downgrade `numpy` if that's easier for you system. 

I'm going to close this issue as resolved since the underlying issue has already been fixed
